lib/src/room.dart

```diff
@@ -1356,8 +1356,28 @@ class Room {
   /// read receipt's location.
   /// If you set `public` to false, only a private receipt will be sent. A private receipt is always sent if `mRead` is set. If no value is provided, the default from the `client` is used.
   /// You can leave out the `eventId`, which will not update the read marker but just send receipts, but there are few cases where that makes sense.
-  Future<void> setReadMarker(String? eventId,
-      {String? mRead, bool? public}) async {
+  Future<void> setReadMarker(
```

Comment:
Can we add some local echo revert for this? Otherwise if we have no network connection, we will mark all rooms as read, but the server won't know this and will push us updated read counts without taking into account the events we have thought we have read already until we open the room again to read the new events, I think.

Reply:
hmm, I know no good solution for this. Any suggestions?

Reply:
No, there is a reason I never implemented this :D 

A 50% solution would be to set the read marker to the previous value if the setReadMarker request fails and the read marker hasn't been updated in the mean time. That won't work when people kill the app during a request, but should work for the common network errors.

The more complicated solution would be to store the pending states in the database and periodically retry while they are pending (they should succeed once we sync again, so the difference in read counts can probably be ignored for one sync). That sounds like quite a bit of work though.

Reply:
The complicated solution could mean that we implement a new feature "ReadMarkerManager" which stores read marker requests in the database, retrys then on failure after every sync similar to the ToDeviceQueue, handles that pending read marker requests are cleared when they come down the sync from another device and so on. You are right, that sounds like a very complicated feature. We could consider this in the future but for now this sounds like too much effort (tests would also be needed) just for this tiny convenience. Therefore I will close this PR for now